src/Avalonia.Controls/Primitives/SelectingItemsControl.cs

```diff
@@ -981,5 +1068,104 @@ private void UpdateFinished()
                 SelectedItems = _updateSelectedItems;
             }
         }
+
+        private class Selection : IEnumerable<int>
+        {
+            private List<int> _list = new List<int>();
```

Comment:
Nit: Both could be `readonly`.

Reply:
Fixed the `readonly`s. `_set` can't be readonly though because it's reassigned.